rebar.config

```diff
@@ -65,7 +65,10 @@
                                         "{{output_dir}}/bin/check_config"},
                                        {copy,
                                        "_build/dev1/lib/aeutils/priv/epoch_config_schema.json",
-                                       "{{output_dir}}/data/epoch_config_schema.json"}]}
+                                       "{{output_dir}}/data/epoch_config_schema.json"},
+                                       {copy,
+                                        "./otp_patches/ebin",
+                                        "{{output_dir}}/patches"}]}
```

Comment:
I see that this approach from `otp_patches` top dir builds using pre-hook in top rebar.config then copies files in release `patches` dir then it relies on `-pa` of vm.args for loading in release. This build&ship approach seems custom: would it be possible to add smoke test in the `package` job for the prod release exercising functionality (e.g. maybe at end of release.py - i.e. after chain has at least one coinbase - start nodes again then call specific HTTP API on one node checking response).

I understand that the point of using `-pa` is aiming at inclusion of patched modules with same name of OTP ones in front of code path: does it work in embedded mode? (I admit I did not test it.)

I am not saying to change approach, though for the sake of maintainability may I suggest we *consider* modularizing otp_patches / compile pre_hook / copy step in release build as separate OTP app? The tricky bit would be whether it is possible to make user app load module with same name of module in OTP app, and make it take precedence in code path.

Reply:
A possible way forward is that the patch system is extended as-needed.

Packaging patches as a separate OTP application would be problematic, as the OTP `systools` script builder doesn't accept duplicate module names, and so would refuse to generate the start script. The `-pa` flag does work with embedded code loading.

One caveat with the chosen approach is that it does require the current OTP version to have source included.